portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,11 +47,34 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Query query = new Query(DATASTORE_LABEL);
         PreparedQuery results = datastore.prepare(query);
 
-        ArrayList<String> commentList = new ArrayList<String>();
-        for (Entity entity : results.asIterable()) {
-            commentList.add((String) entity.getProperty(INPUT_NAME));
+        int numComments;
+        // Get the number of comments the user wants to view. 
+        try {
+            numComments = Integer.parseInt(request.getParameter(NUM_COMMENTS_INPUT));
+        } catch (NumberFormatException e) {
+            numComments = 20;
```

Comment:
nice :-) 

maybe add a private static final int DEFAULT_PAGE_SIZE = 20 and use it here?